homeassistant/components/tplink/__init__.py

```diff
@@ -86,11 +126,43 @@ async def _async_discovery(*_: Any) -> None:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     host = entry.data[CONF_HOST]
+
+    credentials = await get_credentials(hass)
+    connection_type = None
+
+    if connection_type_dict := entry.data.get(CONF_CONNECTION_TYPE):
+        try:
+            connection_type = ConnectionType.from_dict(connection_type_dict)
+        except SmartDeviceException:
+            _LOGGER.warning(
+                "Invalid connection type dict for %s: %s", host, connection_type_dict
+            )
```

Comment:
Should we ever need to construct `ConnectionType` etc. manually instead of simply constructing the `DeviceConfig` from the config data stored inside the config entry?

Reply:
`ConnectionType` serialization now dropped